src/variable_conic_form.jl

```diff
@@ -26,3 +26,12 @@ function conic_form!(x::AbstractVariable, unique_conic_forms::UniqueConicForms)
     end
     return get_conic_form(unique_conic_forms, x)
 end
+
+
+function template(a::AbstractVariable, context)
+    var_inds = get!(context.var_id_to_moi_indices, a.id_hash) do
+        return add_variables!(context.model, a::Variable)
```

Comment:
there are two leaf types: `Constant`s and `AbstractVariable`s. When we reach a variable, we add it to our local state (in `context`) by adding it to the MOI model and to our dictionary. I've used Convex.jl's convention of an explicit hash field, but I think a usual dictionary that does it's own hashing might be cleaner.

Reply:
Comment doesn't make sense as written - did you mean "does its own hashing"?

Reply:
yep, sorry about that. By the way, I am unsure what to do with the hashing stuff in Convex; it's used to avoid reformulating branches of the expression tree if they are identical and have already been formulated, but I don't entirely understand why it works (sometimes the `head` symbol of an atom is hashed in, sometimes it's not; seems like it should be a source of bugs but it hasn't been yet...) and think it should be revamped.